engine/crates/graphql-schema-diff/src/patch/type_definitions.rs

```diff
@@ -0,0 +1,318 @@
+            schema.push_str(changed_field_types[idx].1);
+        } else {
+            schema.push_str(&field.ty().to_string());
+        }
+
+        schema.push('\n');
+    }
+
+    schema.push('}');
+}
+
+fn patch_fields<'a, T>(
+    fields: impl Iterator<Item = FieldDefinition<'a>>,
+    parent: &str,
+    schema: &mut String,
+    paths: &Paths<'_, T>,
+) where
+    T: AsRef<str>,
+{
+    schema.push_str(" {\n");
+
+    let mut changed_field_types = Vec::new();
+    let mut removed_fields = Vec::new();
+
+    for change in paths.iter_second_level(parent) {
+        match change.kind() {
+            ChangeKind::ChangeFieldType => {
+                let field_name = change.second_level().expect("ChangeFieldType without field name");
+                changed_field_types.push((field_name, change.resolved_str()));
+            }
+            ChangeKind::RemoveField => {
+                let field_name = change.second_level().expect("RemoveField without field name");
+                removed_fields.push(field_name);
+            }
+            ChangeKind::AddField => {
+                schema.push_str(INDENTATION);
+                schema.push_str(change.resolved_str().trim());
+                schema.push('\n');
+            }
+            ChangeKind::AddInterfaceImplementation | ChangeKind::RemoveInterfaceImplementation => (), // already handled
+            kind => {
+                debug_assert!(false, "Unhandled change at `{path}`: {kind:?}", path = change.path())
+            }
+        }
+    }
+
+    removed_fields.sort_unstable();
+
+    for field in fields {
+        if removed_fields.binary_search(&field.name()).is_ok() {
+            continue;
+        }
+
+        if let Some(description) = field.description() {
+            let span = description.span();
+            schema.push_str(INDENTATION);
+            schema.push_str(&paths.source()[span.start..span.end]);
+            schema.push('\n');
+        }
+
+        schema.push_str(INDENTATION);
+        schema.push_str(field.name());
+
+        let mut arguments = field.arguments().peekable();
+
+        if arguments.peek().is_some() {
+            schema.push('(');
+
+            while let Some(argument) = arguments.next() {
+                if let Some(description) = argument.description() {
+                    let span = description.span();
+                    schema.push_str(&paths.source()[span.start..span.end]);
+                    schema.push(' ');
+                }
+
+                schema.push_str(argument.name());
+                schema.push_str(": ");
+                schema.push_str(&argument.ty().to_string());
+
+                if argument.default_value().is_some() {
+                    schema.push(' ');
+                    let span = argument.default_value_span();
+                    schema.push_str(&paths.source()[span.start..span.end]);
+                }
+
```

Comment:
Argument directives also intentionally missing here?

Reply:
Same as fields, it won't be complete (diffing for directives isn't there yet), but I'll add the same level of support as in other places, good catch.